crates/sargon/src/signing/testing/interactors/test_sign_interactor.rs

```diff
@@ -2,34 +2,34 @@
 
 use crate::prelude::*;
 
-pub(crate) struct TestSigningParallelInteractor {
-    simulated_user: SimulatedUser,
+pub(crate) struct TestSignInteractor<S: Signable> {
+    pub(crate) simulated_user: SimulatedUser<S>,
 }
 
-impl TestSigningParallelInteractor {
-    pub(crate) fn new(simulated_user: SimulatedUser) -> Self {
+unsafe impl<S: Signable> Sync for SignRequest<S> {}
```

Comment:
Is `unsafe impl<S: Signable> Sync ` needed? 

What if you constrain `S` to not only be `Signable` but also `Send + Sync`? like so:

```rust
pub struct SignRequest<S: Signable + Send + Sync> {...}
```
?

Reply:
(which will require you to update some impl blocks too...)

Reply:
Does not seem to work

Reply:
lets revisit this. Dont resolve this please. This is production code. We dont wanna do `unsafe impl Send` in production code. Can you paste the compilation errpr?

Reply:
> We dont wanna do unsafe impl Send in production code.
This is a test interactor no?

Reply:
`SignRequest` is not test only type. It is production type right? But you mean this whole **file** is gated behind `cfg(test)`? :) if so, yes then it is ok, but to make it even more clear I would do this:

```diff
+ /// N.B. We only unsafe impl `Sync` for test (even though the type `SignRequest` itself is not test only).
+ #[cfg(test)]
unsafe impl<S: Signable> Sync for SignRequest<S> {}
```

Reply:
still can you paste the compilation error you get without it?

Reply:
```
cargo check
   Compiling sargon v1.1.72 (/Users/micbakos/Developer/rdx/sargon/crates/sargon)
error[E0277]: `S` cannot be sent between threads safely
   --> crates/sargon/src/signing/collector/signatures_collector.rs:213:22
    |
213 |                 self.request_for_parallel_interactor(factor_sources_of_kind);
    |                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ `S` cannot be sent between threads safely
    |
note: required by a bound in `sign_request::SignRequest`
   --> crates/sargon/src/signing/host_interaction/sign_request.rs:4:38
    |
4   | pub struct SignRequest<S: Signable + Send + Sync> {
    |                                      ^^^^ required by this bound in `SignRequest`
help: consider further restricting this bound
    |
114 | impl<S: Signable + std::marker::Send> SignaturesCollector<S> {
    |                  +++++++++++++++++++

error[E0277]: `S` cannot be shared between threads safely
   --> crates/sargon/src/signing/collector/signatures_collector.rs:213:22
    |
213 |                 self.request_for_parallel_interactor(factor_sources_of_kind);
    |                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ `S` cannot be shared between threads safely
    |
note: required by a bound in `sign_request::SignRequest`
   --> crates/sargon/src/signing/host_interaction/sign_request.rs:4:45
    |
4   | pub struct SignRequest<S: Signable + Send + Sync> {
    |                                             ^^^^ required by this bound in `SignRequest`
help: consider further restricting this bound
    |
114 | impl<S: Signable + std::marker::Sync> SignaturesCollector<S> {
    |                  +++++++++++++++++++

error[E0277]: `S` cannot be sent between threads safely
   --> crates/sargon/src/signing/collector/signatures_collector.rs:221:68
    |
221 |             debug!("Dispatching poly request to interactor: {:?}", request);
    |                                                             ----   ^^^^^^^ `S` cannot be sent between threads safely
    |                                                             |
    |                                                             required by a bound introduced by this call
    |
note: required for `sign_request::SignRequest<S>` to implement `derive_more::Debug`
   --> crates/sargon/src/signing/host_interaction/sign_request.rs:3:10
    |
3   | #[derive(Debug, Clone, PartialEq, Eq)]
    |          ^^^^^
4   | pub struct SignRequest<S: Signable + Send + Sync> {
    |                                      ---- unsatisfied trait bound introduced in this `derive` macro
note: required by a bound in `core::fmt::rt::Argument::<'a>::new_debug`
   --> /Users/micbakos/.rustup/toolchains/nightly-2024-07-30-aarch64-apple-darwin/lib/rustlib/src/rust/library/core/src/fmt/rt.rs:117:29
    |
117 |     pub fn new_debug<'b, T: Debug>(x: &'b T) -> Argument<'_> {
    |                             ^^^^^ required by this bound in `Argument::<'a>::new_debug`
    = note: this error originates in the macro `$crate::__private_api::format_args` which comes from the expansion of the derive macro `Debug` (in Nightly builds, run with -Z macro-backtrace for more info)
help: consider further restricting this bound
    |
114 | impl<S: Signable + std::marker::Send> SignaturesCollector<S> {
    |                  +++++++++++++++++++

error[E0277]: `S` cannot be shared between threads safely
   --> crates/sargon/src/signing/collector/signatures_collector.rs:221:68
    |
221 |             debug!("Dispatching poly request to interactor: {:?}", request);
    |                                                             ----   ^^^^^^^ `S` cannot be shared between threads safely
    |                                                             |
    |                                                             required by a bound introduced by this call
    |
note: required for `sign_request::SignRequest<S>` to implement `derive_more::Debug`
   --> crates/sargon/src/signing/host_interaction/sign_request.rs:3:10
    |
3   | #[derive(Debug, Clone, PartialEq, Eq)]
    |          ^^^^^
4   | pub struct SignRequest<S: Signable + Send + Sync> {
    |                                             ---- unsatisfied trait bound introduced in this `derive` macro
note: required by a bound in `core::fmt::rt::Argument::<'a>::new_debug`
   --> /Users/micbakos/.rustup/toolchains/nightly-2024-07-30-aarch64-apple-darwin/lib/rustlib/src/rust/library/core/src/fmt/rt.rs:117:29
    |
117 |     pub fn new_debug<'b, T: Debug>(x: &'b T) -> Argument<'_> {
    |                             ^^^^^ required by this bound in `Argument::<'a>::new_debug`
    = note: this error originates in the macro `$crate::__private_api::format_args` which comes from the expansion of the derive macro `Debug` (in Nightly builds, run with -Z macro-backtrace for more info)
help: consider further restricting this bound
    |
114 | impl<S: Signable + std::marker::Sync> SignaturesCollector<S> {
    |                  +++++++++++++++++++

error[E0277]: `S` cannot be sent between threads safely
   --> crates/sargon/src/signing/collector/signatures_collector.rs:222:57
    |
222 |             let response = self.dependencies.interactor.sign(request).await;
    |                                                         ^^^^ `S` cannot be sent between threads safely
    |
note: required by a bound in `sign_request::SignRequest`
   --> crates/sargon/src/signing/host_interaction/sign_request.rs:4:38
    |
4   | pub struct SignRequest<S: Signable + Send + Sync> {
    |                                      ^^^^ required by this bound in `SignRequest`
help: consider further restricting this bound
    |
114 | impl<S: Signable + std::marker::Send> SignaturesCollector<S> {
    |                  +++++++++++++++++++

error[E0277]: `S` cannot be shared between threads safely
   --> crates/sargon/src/signing/collector/signatures_collector.rs:222:57
    |
222 |             let response = self.dependencies.interactor.sign(request).await;
    |                                                         ^^^^ `S` cannot be shared between threads safely
    |
note: required by a bound in `sign_request::SignRequest`
   --> crates/sargon/src/signing/host_interaction/sign_request.rs:4:45
    |
4   | pub struct SignRequest<S: Signable + Send + Sync> {
    |                                             ^^^^ required by this bound in `SignRequest`
help: consider further restricting this bound
    |
114 | impl<S: Signable + std::marker::Sync> SignaturesCollector<S> {
    |                  +++++++++++++++++++

error[E0277]: `S` cannot be sent between threads safely
   --> crates/sargon/src/signing/collector/signatures_collector.rs:236:26
    |
236 |                     self.request_for_serial_interactor(kind, &factor_source_id);
    |                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ `S` cannot be sent between threads safely
    |
note: required by a bound in `sign_request::SignRequest`
   --> crates/sargon/src/signing/host_interaction/sign_request.rs:4:38
    |
4   | pub struct SignRequest<S: Signable + Send + Sync> {
    |                                      ^^^^ required by this bound in `SignRequest`
help: consider further restricting this bound
    |
114 | impl<S: Signable + std::marker::Send> SignaturesCollector<S> {
    |                  +++++++++++++++++++

error[E0277]: `S` cannot be shared between threads safely
   --> crates/sargon/src/signing/collector/signatures_collector.rs:236:26
    |
236 |                     self.request_for_serial_interactor(kind, &factor_source_id);
    |                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ `S` cannot be shared between threads safely
    |
note: required by a bound in `sign_request::SignRequest`
   --> crates/sargon/src/signing/host_interaction/sign_request.rs:4:45
    |
4   | pub struct SignRequest<S: Signable + Send + Sync> {
    |                                             ^^^^ required by this bound in `SignRequest`
help: consider further restricting this bound
    |
114 | impl<S: Signable + std::marker::Sync> SignaturesCollector<S> {
    |                  +++++++++++++++++++

error[E0277]: `S` cannot be sent between threads safely
   --> crates/sargon/src/signing/collector/signatures_collector.rs:246:72
    |
246 |                 debug!("Dispatching mono request to interactor: {:?}", request);
    |                                                                 ----   ^^^^^^^ `S` cannot be sent between threads safely
    |                                                                 |
    |                                                                 required by a bound introduced by this call
    |
note: required for `sign_request::SignRequest<S>` to implement `derive_more::Debug`
   --> crates/sargon/src/signing/host_interaction/sign_request.rs:3:10
    |
3   | #[derive(Debug, Clone, PartialEq, Eq)]
    |          ^^^^^
4   | pub struct SignRequest<S: Signable + Send + Sync> {
    |                                      ---- unsatisfied trait bound introduced in this `derive` macro
note: required by a bound in `core::fmt::rt::Argument::<'a>::new_debug`
   --> /Users/micbakos/.rustup/toolchains/nightly-2024-07-30-aarch64-apple-darwin/lib/rustlib/src/rust/library/core/src/fmt/rt.rs:117:29
    |
117 |     pub fn new_debug<'b, T: Debug>(x: &'b T) -> Argument<'_> {
    |                             ^^^^^ required by this bound in `Argument::<'a>::new_debug`
    = note: this error originates in the macro `$crate::__private_api::format_args` which comes from the expansion of the derive macro `Debug` (in Nightly builds, run with -Z macro-backtrace for more info)
help: consider further restricting this bound
    |
114 | impl<S: Signable + std::marker::Send> SignaturesCollector<S> {
    |                  +++++++++++++++++++

error[E0277]: `S` cannot be shared between threads safely
   --> crates/sargon/src/signing/collector/signatures_collector.rs:246:72
    |
246 |                 debug!("Dispatching mono request to interactor: {:?}", request);
    |                                                                 ----   ^^^^^^^ `S` cannot be shared between threads safely
    |                                                                 |
    |                                                                 required by a bound introduced by this call
    |
note: required for `sign_request::SignRequest<S>` to implement `derive_more::Debug`
   --> crates/sargon/src/signing/host_interaction/sign_request.rs:3:10
    |
3   | #[derive(Debug, Clone, PartialEq, Eq)]
    |          ^^^^^
4   | pub struct SignRequest<S: Signable + Send + Sync> {
    |                                             ---- unsatisfied trait bound introduced in this `derive` macro
note: required by a bound in `core::fmt::rt::Argument::<'a>::new_debug`
   --> /Users/micbakos/.rustup/toolchains/nightly-2024-07-30-aarch64-apple-darwin/lib/rustlib/src/rust/library/core/src/fmt/rt.rs:117:29
    |
117 |     pub fn new_debug<'b, T: Debug>(x: &'b T) -> Argument<'_> {
    |                             ^^^^^ required by this bound in `Argument::<'a>::new_debug`
    = note: this error originates in the macro `$crate::__private_api::format_args` which comes from the expansion of the derive macro `Debug` (in Nightly builds, run with -Z macro-backtrace for more info)
help: consider further restricting this bound
    |
114 | impl<S: Signable + std::marker::Sync> SignaturesCollector<S> {
    |                  +++++++++++++++++++

error[E0277]: `S` cannot be sent between threads safely
   --> crates/sargon/src/signing/collector/signatures_collector.rs:248:61
    |
248 |                 let response = self.dependencies.interactor.sign(request).await;
    |                                                             ^^^^ `S` cannot be sent between threads safely
    |
note: required by a bound in `sign_request::SignRequest`
   --> crates/sargon/src/signing/host_interaction/sign_request.rs:4:38
    |
4   | pub struct SignRequest<S: Signable + Send + Sync> {
    |                                      ^^^^ required by this bound in `SignRequest`
help: consider further restricting this bound
    |
114 | impl<S: Signable + std::marker::Send> SignaturesCollector<S> {
    |                  +++++++++++++++++++

error[E0277]: `S` cannot be shared between threads safely
   --> crates/sargon/src/signing/collector/signatures_collector.rs:248:61
    |
248 |                 let response = self.dependencies.interactor.sign(request).await;
    |                                                             ^^^^ `S` cannot be shared between threads safely
    |
note: required by a bound in `sign_request::SignRequest`
   --> crates/sargon/src/signing/host_interaction/sign_request.rs:4:45
    |
4   | pub struct SignRequest<S: Signable + Send + Sync> {
    |                                             ^^^^ required by this bound in `SignRequest`
help: consider further restricting this bound
    |
114 | impl<S: Signable + std::marker::Sync> SignaturesCollector<S> {
    |                  +++++++++++++++++++

error[E0277]: `S` cannot be sent between threads safely
   --> crates/sargon/src/signing/collector/signatures_collector.rs:297:10
    |
297 |     ) -> SignRequest<S> {
    |          ^^^^^^^^^^^^^^ `S` cannot be sent between threads safely
    |
note: required by a bound in `sign_request::SignRequest`
   --> crates/sargon/src/signing/host_interaction/sign_request.rs:4:38
    |
4   | pub struct SignRequest<S: Signable + Send + Sync> {
    |                                      ^^^^ required by this bound in `SignRequest`
help: consider further restricting this bound
    |
114 | impl<S: Signable + std::marker::Send> SignaturesCollector<S> {
    |                  +++++++++++++++++++

error[E0277]: `S` cannot be shared between threads safely
   --> crates/sargon/src/signing/collector/signatures_collector.rs:297:10
    |
297 |     ) -> SignRequest<S> {
    |          ^^^^^^^^^^^^^^ `S` cannot be shared between threads safely
    |
note: required by a bound in `sign_request::SignRequest`
   --> crates/sargon/src/signing/host_interaction/sign_request.rs:4:45
    |
4   | pub struct SignRequest<S: Signable + Send + Sync> {
    |                                             ^^^^ required by this bound in `SignRequest`
help: consider further restricting this bound
    |
114 | impl<S: Signable + std::marker::Sync> SignaturesCollector<S> {
    |                  +++++++++++++++++++

error[E0277]: `S` cannot be sent between threads safely
   --> crates/sargon/src/signing/collector/signatures_collector.rs:317:10
    |
317 |     ) -> SignRequest<S> {
    |          ^^^^^^^^^^^^^^ `S` cannot be sent between threads safely
    |
note: required by a bound in `sign_request::SignRequest`
   --> crates/sargon/src/signing/host_interaction/sign_request.rs:4:38
    |
4   | pub struct SignRequest<S: Signable + Send + Sync> {
    |                                      ^^^^ required by this bound in `SignRequest`
help: consider further restricting this bound
    |
114 | impl<S: Signable + std::marker::Send> SignaturesCollector<S> {
    |                  +++++++++++++++++++

error[E0277]: `S` cannot be shared between threads safely
   --> crates/sargon/src/signing/collector/signatures_collector.rs:317:10
    |
317 |     ) -> SignRequest<S> {
    |          ^^^^^^^^^^^^^^ `S` cannot be shared between threads safely
    |
note: required by a bound in `sign_request::SignRequest`
   --> crates/sargon/src/signing/host_interaction/sign_request.rs:4:45
    |
4   | pub struct SignRequest<S: Signable + Send + Sync> {
    |                                             ^^^^ required by this bound in `SignRequest`
help: consider further restricting this bound
    |
114 | impl<S: Signable + std::marker::Sync> SignaturesCollector<S> {
    |                  +++++++++++++++++++

error[E0277]: `S` cannot be sent between threads safely
 --> crates/sargon/src/signing/host_interaction/sign_interactor.rs:7:18
  |
7 |         request: SignRequest<S>,
  |                  ^^^^^^^^^^^^^^ `S` cannot be sent between threads safely
  |
note: required by a bound in `sign_request::SignRequest`
 --> crates/sargon/src/signing/host_interaction/sign_request.rs:4:38
  |
4 | pub struct SignRequest<S: Signable + Send + Sync> {
  |                                      ^^^^ required by this bound in `SignRequest`
help: consider further restricting this bound
  |
4 | pub trait SignInteractor<S: Signable + std::marker::Send>: Send + Sync {
  |                                      +++++++++++++++++++

error[E0277]: `S` cannot be shared between threads safely
 --> crates/sargon/src/signing/host_interaction/sign_interactor.rs:7:18
  |
7 |         request: SignRequest<S>,
  |                  ^^^^^^^^^^^^^^ `S` cannot be shared between threads safely
  |
note: required by a bound in `sign_request::SignRequest`
 --> crates/sargon/src/signing/host_interaction/sign_request.rs:4:45
  |
4 | pub struct SignRequest<S: Signable + Send + Sync> {
  |                                             ^^^^ required by this bound in `SignRequest`
help: consider further restricting this bound
  |
4 | pub trait SignInteractor<S: Signable + std::marker::Sync>: Send + Sync {
  |                                      +++++++++++++++++++

error[E0277]: `S` cannot be sent between threads safely
   --> crates/sargon/src/signing/collector/signatures_collector.rs:309:13
    |
307 |         SignRequest::new(
    |         ---------------- required by a bound introduced by this call
308 |             factor_source_kind,
309 |             IndexMap::just((*factor_source_id, batch_signing_request)),
    |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ `S` cannot be sent between threads safely
    |
note: required by a bound in `sign_request::SignRequest::<S>::new`
   --> crates/sargon/src/signing/host_interaction/sign_request.rs:20:20
    |
20  | impl<S: Signable + Send + Sync> SignRequest<S> {
    |                    ^^^^ required by this bound in `SignRequest::<S>::new`
...
25  |     pub(crate) fn new(
    |                   --- required by a bound in this associated function
help: consider further restricting this bound
    |
114 | impl<S: Signable + std::marker::Send> SignaturesCollector<S> {
    |                  +++++++++++++++++++

error[E0277]: `S` cannot be shared between threads safely
   --> crates/sargon/src/signing/collector/signatures_collector.rs:309:13
    |
307 |         SignRequest::new(
    |         ---------------- required by a bound introduced by this call
308 |             factor_source_kind,
309 |             IndexMap::just((*factor_source_id, batch_signing_request)),
    |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ `S` cannot be shared between threads safely
    |
note: required by a bound in `sign_request::SignRequest::<S>::new`
   --> crates/sargon/src/signing/host_interaction/sign_request.rs:20:27
    |
20  | impl<S: Signable + Send + Sync> SignRequest<S> {
    |                           ^^^^ required by this bound in `SignRequest::<S>::new`
...
25  |     pub(crate) fn new(
    |                   --- required by a bound in this associated function
help: consider further restricting this bound
    |
114 | impl<S: Signable + std::marker::Sync> SignaturesCollector<S> {
    |                  +++++++++++++++++++

error[E0277]: `S` cannot be sent between threads safely
   --> crates/sargon/src/signing/collector/signatures_collector.rs:307:9
    |
307 | /         SignRequest::new(
308 | |             factor_source_kind,
309 | |             IndexMap::just((*factor_source_id, batch_signing_request)),
310 | |             invalid_transactions_if_neglected,
311 | |         )
    | |_________^ `S` cannot be sent between threads safely
    |
note: required by a bound in `sign_request::SignRequest`
   --> crates/sargon/src/signing/host_interaction/sign_request.rs:4:38
    |
4   | pub struct SignRequest<S: Signable + Send + Sync> {
    |                                      ^^^^ required by this bound in `SignRequest`
help: consider further restricting this bound
    |
114 | impl<S: Signable + std::marker::Send> SignaturesCollector<S> {
    |                  +++++++++++++++++++

error[E0277]: `S` cannot be shared between threads safely
   --> crates/sargon/src/signing/collector/signatures_collector.rs:307:9
    |
307 | /         SignRequest::new(
308 | |             factor_source_kind,
309 | |             IndexMap::just((*factor_source_id, batch_signing_request)),
310 | |             invalid_transactions_if_neglected,
311 | |         )
    | |_________^ `S` cannot be shared between threads safely
    |
note: required by a bound in `sign_request::SignRequest`
   --> crates/sargon/src/signing/host_interaction/sign_request.rs:4:45
    |
4   | pub struct SignRequest<S: Signable + Send + Sync> {
    |                                             ^^^^ required by this bound in `SignRequest`
help: consider further restricting this bound
    |
114 | impl<S: Signable + std::marker::Sync> SignaturesCollector<S> {
    |                  +++++++++++++++++++

error[E0277]: `S` cannot be sent between threads safely
   --> crates/sargon/src/signing/collector/signatures_collector.rs:344:13
    |
342 |         SignRequest::new(
    |         ---------------- required by a bound introduced by this call
343 |             factor_sources_of_kind.kind,
344 |             per_factor_source,
    |             ^^^^^^^^^^^^^^^^^ `S` cannot be sent between threads safely
    |
note: required by a bound in `sign_request::SignRequest::<S>::new`
   --> crates/sargon/src/signing/host_interaction/sign_request.rs:20:20
    |
20  | impl<S: Signable + Send + Sync> SignRequest<S> {
    |                    ^^^^ required by this bound in `SignRequest::<S>::new`
...
25  |     pub(crate) fn new(
    |                   --- required by a bound in this associated function
help: consider further restricting this bound
    |
114 | impl<S: Signable + std::marker::Send> SignaturesCollector<S> {
    |                  +++++++++++++++++++

error[E0277]: `S` cannot be shared between threads safely
   --> crates/sargon/src/signing/collector/signatures_collector.rs:344:13
    |
342 |         SignRequest::new(
    |         ---------------- required by a bound introduced by this call
343 |             factor_sources_of_kind.kind,
344 |             per_factor_source,
    |             ^^^^^^^^^^^^^^^^^ `S` cannot be shared between threads safely
    |
note: required by a bound in `sign_request::SignRequest::<S>::new`
   --> crates/sargon/src/signing/host_interaction/sign_request.rs:20:27
    |
20  | impl<S: Signable + Send + Sync> SignRequest<S> {
    |                           ^^^^ required by this bound in `SignRequest::<S>::new`
...
25  |     pub(crate) fn new(
    |                   --- required by a bound in this associated function
help: consider further restricting this bound
    |
114 | impl<S: Signable + std::marker::Sync> SignaturesCollector<S> {
    |                  +++++++++++++++++++

error[E0277]: `S` cannot be sent between threads safely
   --> crates/sargon/src/signing/collector/signatures_collector.rs:342:9
    |
342 | /         SignRequest::new(
343 | |             factor_sources_of_kind.kind,
344 | |             per_factor_source,
345 | |             invalid_transactions_if_neglected,
346 | |         )
    | |_________^ `S` cannot be sent between threads safely
    |
note: required by a bound in `sign_request::SignRequest`
   --> crates/sargon/src/signing/host_interaction/sign_request.rs:4:38
    |
4   | pub struct SignRequest<S: Signable + Send + Sync> {
    |                                      ^^^^ required by this bound in `SignRequest`
help: consider further restricting this bound
    |
114 | impl<S: Signable + std::marker::Send> SignaturesCollector<S> {
    |                  +++++++++++++++++++

error[E0277]: `S` cannot be shared between threads safely
   --> crates/sargon/src/signing/collector/signatures_collector.rs:342:9
    |
342 | /         SignRequest::new(
343 | |             factor_sources_of_kind.kind,
344 | |             per_factor_source,
345 | |             invalid_transactions_if_neglected,
346 | |         )
    | |_________^ `S` cannot be shared between threads safely
    |
note: required by a bound in `sign_request::SignRequest`
   --> crates/sargon/src/signing/host_interaction/sign_request.rs:4:45
    |
4   | pub struct SignRequest<S: Signable + Send + Sync> {
    |                                             ^^^^ required by this bound in `SignRequest`
help: consider further restricting this bound
    |
114 | impl<S: Signable + std::marker::Sync> SignaturesCollector<S> {
    |                  +++++++++++++++++++

For more information about this error, try `rustc --explain E0277`.
error: could not compile `sargon` (lib) due to 26 previous errors
```

Reply:
did you try what was suggested?

```
help: consider further restricting this bound
    |
114 | impl<S: Signable + std::marker::Send> SignaturesCollector<S> {
    |                  +++++++++++++++++++

help: consider further restricting this bound
    |
114 | impl<S: Signable + std::marker::Sync> SignaturesCollector<S> {
```

Does that fix it?

Reply:
No

Reply:
@CyonAlexRDX Do you have any other suggestions on how this could be resolved?

Reply:
The combination of the above + removing the unsafe impl for SignRequest, compiles and tests pass.

I cannot remove the unsafe impl for SimulatedUser though.

Reply:
I think;

1. Signable: ... + Send + Sync
2. Signable::ID: ... + Send + Sync
3. Signable::Payload: ... + Send + Sync

Look ok!

Reply:
100% OK to have it on `SimulatedUser`!